sql/solutions/lesson4.md

```diff
@@ -0,0 +1,125 @@
+---
+layout: page
+title: Introduction to SQL Solutions
+---
+# Lesson 1 exercises Solutions
+
+## L4.1 Update the age of the person named Luis Marco Polo.
+
+Solution 1:
+First find person's id:
+```SQL
+SELECT *
+  FROM people
+ WHERE name = 'Luis Marco Polo';
+```
+Update age using id number:
+```SQL
+UPDATE people
+   SET age = age + 1
+ WHERE id = 3;
+```
+
+Solution 2:
+Just use their name as identifier to update age. (When could this cause problems?)
+```SQL
+UPDATE people
+   SET age = age + 1
+ WHERE name = 'Luis Marco Polo';
```

Comment:
This could cause problems indeed when the name is not UNIQUE, which normally can happen... how many Oliver Smith are there? :)

I'm ok to leave the solution 2 anyway, but you can add the warning about the risk that this method may face.

Reply:
Solution 1 has 2 steps: getting individuals id first, then updating age using id number as identifier.

Solution 2 I put in more because a student might actually write that query -- and in this case get the correct answer. But I put a discussion question: 'When could this cause problems?'

Reply:
But yes, solution 2 is a problem. I'll go add a warning. :)